### PR TITLE
Transform early

### PR DESCRIPTION
**Pull request by a3mitskevich**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,16 +42,29 @@
   "author": "Aleksandr Mitskevich",
   "license": "MIT",
   "dependencies": {
-    "fontext": "1.0.10",
-    "lodash.groupby": "4.6.0"
+    "css-tree": "2.3.1",
+    "fontext": "1.1.2",
+    "lodash.groupby": "4.6.0",
+    "picocolors": "^1.0.0"
```
Comment (PlisMikhail) [inline package.json:48]: You dont need `^` here

File: package.json [hunk 2]
```diff
@@ -31,16 +42,29 @@
...
   },
   "peerDependencies": {
     "vite": "^4.0.0 || ^5.0.0"
   },
   "devDependencies": {
+    "@tsconfig/node20": "20.1.2",
+    "@types/css-tree": "2.3.5",
+    "@types/jest": "^29.5.11",
```
Comment (PlisMikhail) [inline package.json:56]: You dont need `^` here

File: src/analyser.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    const extension = getExtension<keyof ImportResolvers>(id)
+    return await this.resolvers[extension]?.(id, importer) ??
+        await this.resolvers.sass(id, importer) ??
+        await this.resolvers.less(id, importer) ??
+        id
+  }
+
+  process = async (code: string, fontNames: string[], importer?: string): Promise<FontFaceMeta[]> => {
+    const cssFileMeta = await this.generateCssFileMeta(code, fontNames, importer)
+    const foundedFountNames = cssFileMeta.fonts.map(font => font.name)
```
Comment (PlisMikhail) [inline src/analyser.ts:58]: `foundedFountNames` you mean `foundedFontNames`?

File: src/analyser.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+  findAtRules = (ast: CssNode): TargetAttributes => {
+    const nodes = Object.fromEntries<CssNode[]>(
+      Object.values(NODE_TYPE).map(type => [type, []]),
+    ) as TargetAttributes
+
+    walk(ast, {
+      visit: 'Atrule',
+      enter (this: WalkContext, node: Atrule) {
+        if (node.name === NODE_TYPE.IMPORT || node.name === NODE_TYPE.FONT_FACE) {
```
Comment (PlisMikhail) [inline src/analyser.ts:40]: what about `@mixin` ?

**Review by PlisMikhail — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (a3mitskevich) [reply package.json]: removed

File: src/analyser.ts (hunk 3 above)
Comment (a3mitskevich) [reply src/analyser.ts]: renamed

File: src/analyser.ts (hunk 4 above)
Comment (a3mitskevich) [reply src/analyser.ts]: good point. Thank you for your attention. Reworked to another solution

File: package.json [hunk 5]
```diff
@@ -31,16 +42,31 @@
...
   },
   "peerDependencies": {
     "vite": "^4.0.0 || ^5.0.0"
   },
   "devDependencies": {
-    "@types/lodash.groupby": "4.6.7",
-    "@types/node": "20.4.5",
+    "@tsconfig/node20": "20.1.2",
+    "@types/jest": "29.5.11",
+    "@types/lodash.camelcase": "^4.3.9",
```
Comment (PlisMikhail) [inline package.json:57]: redundant `^`

File: package.json [hunk 6]
```diff
@@ -31,16 +42,31 @@
...
+    "@types/lodash.groupby": "^4.6.9",
```
Comment (PlisMikhail) [inline package.json:58]: redundant `^`

File: src/cache.ts [hunk 7]
```diff
@@ -1,43 +1,59 @@
...
+  createDir (): void {
+    if (this.exist) {
+      return
+    }
+    mkdirSync(this.path, { recursive: true })
+  }
+
+  clearCache (pattern?: string): void {
+    const remove = (target: string, recursive: boolean = false): void => {
+      console.log('remove', target)
```
Comment (PlisMikhail) [inline src/cache.ts:40]: do you still need this?

**Review by PlisMikhail — CHANGES_REQUESTED** [review]

File: src/cache.ts (hunk 7 above)
Comment (a3mitskevich) [reply src/cache.ts]: removed

**Review by PlisMikhail — APPROVED** [review]

**Pull request merged**